src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,259 @@
+"""Implementation of Rule L064."""
+
+from typing import Optional
+
+import regex
+
+from sqlfluff.core.parser.segments.raw import CodeSegment
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+)
+
+
+@document_configuration
+@document_fix_compatible
+class Rule_L064(BaseRule):
+    r"""Consistent usage of preferred quotes for quoted literals.
+
+    Some databases allow single quotes as well as double quotes to express a quoted
+    literal. Prefer one type of quotes as specified in rule setting. Falling back to
+    alternate quotes to reduce escapes.
+
+    Dollar quoted raw strings are excluded from this rule, as they are mostly used for
+    literal UDF Body definitions.
+
+    .. note::
+       This rule only checks quoted literals and not quoted identifiers as they often
+       cannot interchange single and double quotes
+
+       For dialects that do not support single *and* double quotes for quoted literals
+       automated fixes can potentially break working SQL code.
+
+       This rule is only enabled for dialects that allow single *and* double quotes for
+       quoted literals (currently ``bigquery``, ``hive``, ``mysql``, ``sparksql``).
+       It can be enabled for other dialects with the ``force_enable = True`` flag.
+
+    **Anti-pattern**
+
+    .. code-block:: sql
+       :force:
+
+        select
+            "abc",
+            'abc',
+            "\"",
+            "abc" = 'abc'
+        from foo
+
+    **Best practice**
+
+    Ensure all quoted literals use preferred quotes, unless escaping can be reduced by
+    using alternate quotes.
+
+    .. code-block:: sql
+       :force:
+
+        select
+            "abc",
+            "abc",
+            '"',
+            "abc" = "abc"
+        from foo
+
+    """
+
+    config_keywords = ["preferred_string_quotes", "force_enable"]
+    _dialects_with_double_quoted_strings = [
+        "bigquery",
+        "hive",
+        "mysql",
+        "sparksql",
+    ]
+
+    _quotes_mapping = {
+        "single_quotes": {
+            "common_name": "single quotes",
+            "preferred_quote_char": "'",
+            "alternate_quote_char": '"',
+        },
+        "double_quotes": {
+            "common_name": "double quotes",
+            "preferred_quote_char": '"',
+            "alternate_quote_char": "'",
+        },
+    }
+    # BigQuery string prefix characters.
+    _string_prefix_chars = "rbRB"
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        # Config type hints
+        self.preferred_string_quotes: str
+        self.force_enable: bool
+
+        if (
+            context.dialect.name not in self._dialects_with_double_quoted_strings
+            and not self.force_enable
```

Comment:
Puts the fast flag check first, also tweaks the logic to eliminate use of `not` (easier to read IMHO). Also, I suggest we move this `if` below the one that checks the segment name, because the segment name check is faster. (We like `_eval()` functions that exit quickly when there's nothing to do.)
```suggestion
            self.force_enable or
            context.dialect.name in self._dialects_with_double_quoted_strings
```